PTerm-UI/TerminalEmulator.class.st

```diff
@@ -345,13 +360,6 @@ TerminalEmulator >> browseWorkingDirectory [
 	[FileList openOn: self workingDirectory] on: Error do:[self cwdNotFound ]
 ]
 
-{ #category : #connections }
-TerminalEmulator >> closeConnection [
-
-	(self confirm: 'Really disconnect?') ifTrue: [tty session endpoint close. ^true].
```

Comment:
Not sure it is a good idea to remove this check, as there maybe some important tasks running (i.e. installing app, sudoing command modifying system etc.). 

Accidentally close these stasks may corrupt the current system

Reply:
I prefer having the confirmation dialog as well. Though I would like to suppress it for some processes as can be done with the setting [“Ask before closing”](https://support.apple.com/en-gb/guide/terminal/trmlshll/2.13/mac/13.0) in Terminal on macOS (see the screenshot below). When I open a Terminal window and run `sleep 10`, it will ask for confirmation first when I click the window’s close button before the ‘sleep’ process is done. But if I run `screen sleep 10`, it does not ask for confirmation. The list only applies to the direct child processes of the login shell process. So if I run `sh -c 'screen sleep 10; echo done'`, it still asks for confirmation.

![Terminal Setting "Ask Before Closing"](https://github.com/lxsang/PTerm/assets/1611248/b4ba5e72-3422-46cd-bc70-c5f327230710)

Reply:
For now we keep the current behavior on this PR, i open a new enhancement issue on @Rinzwind  comment. We should continue our discussion on this topic on that issue